Headers/DebugServer2/Host/File.h

```diff
@@ -0,0 +1,58 @@
+//
+// Copyright (c) 2014-present, Facebook, Inc.
+// All rights reserved.
+//
+// This source code is licensed under the University of Illinois/NCSA Open
+// Source License found in the LICENSE file in the root directory of this
+// source tree. An additional grant of patent rights can be found in the
+// PATENTS file in the same directory.
+//
+
+#ifndef __DebugServer2_Host_File_h
+#define __DebugServer2_Host_File_h
+
+#include "DebugServer2/Base.h"
+#include "DebugServer2/Constants.h"
+#include "DebugServer2/Utils/Log.h"
+
+#include <string>
+
+namespace ds2 {
+namespace Host {
+
+class File {
+public:
+  File(std::string const &path, uint32_t flags, uint32_t mode);
+  ~File();
+
+public:
+  File(const File &other) = delete;
+  File &operator=(const File &other) = delete;
+
+public:
+  File(File &&other) { *this = std::move(other); }
+
+  File &operator=(File &&other) {
```

Comment:
Yeah otherwise you'll never `close()` the current FD you own.

On Mon, Jun 13, 2016 at 2:32 PM, Francis Ricci <notifications@github.com>
wrote:

> In Headers/DebugServer2/Host/File.h
> <https://github.com/facebook/ds2/pull/353#discussion_r66872451>:
>
> > +namespace ds2 {
> > +namespace Host {
> > +
> > +class File {
> > +public:
> > +  File(std::string const &path, uint32_t flags, uint32_t mode);
> > +  ~File();
> > +
> > +public:
> > +  File(const File &other) = delete;
> > +  File &operator=(const File &other) = delete;
> > +
> > +public:
> > +  File(File &&other) { *this = std::move(other); }
> > +
> > +  File &operator=(File &&other) {
>
> Didn't know I had to do that manually. But yeah, makes sense.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/facebook/ds2/pull/353/files/e78f7a117788527571efd48e401504161ff5090d..d75dab9690e2408866ce63d40f6d87e726e7de3e#r66872451>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AADHbKpkZ9Z6AeTt9xusMqALR-bFXpicks5qLcx1gaJpZM4ImqRQ>
> .
>



-- 
Stephane Sezer